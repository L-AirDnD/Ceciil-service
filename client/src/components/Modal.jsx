import React from 'react';

const { styled } = window;

const Body = styled.div`
  display: flex;
  left: 0px;
  top: 0px;
  width: 100%;
  height: 100%;
  background: rgba(0, 0, 0, 0.85);
  justify-content: center;
  align-items: center;
`;

const Container = styled.div`
  width: 100%;
`;

const TopBar = styled.div`
  width: 100%;
  height: 86px;
`;

const Exit = styled.div`
  float: right;
  right: 0px;
  width: 50px;
  height: 50px;
  margin-top: 13px;
  color: white;
  font-size: 28px;
  cursor: pointer;
`;

const Photo = styled.div`
  display: flex;
  width: 100%;
  height: 510px;
  margin-left: auto;
  margin-right: auto;
  align-content: center;
`;

const Image = styled.img`
  min-width: 400px;
  max-width: 800px;
  max-height: 500px;
  width: 100%;
  height: 100%;
  margin-left: auto;
  margin-right: auto;
  cursor: pointer;
`;

const Arrow = styled.div`
  display: table;
  width 100px;
  padding: 3px;
  color: white;
  font-size: 42px;
  cursor: pointer;
`;

const Right = styled(Arrow)`
  display: table-cell; 
  padding-top: 280px;
  text-align: center; 
  vertical-align: middle; 
`;

const Left = styled(Arrow)`
  display: table-cell; 
  padding-top: 280px;
  text-align: center; 
  vertical-align: middle; 
`;

const BottomBar = styled.div`
  width: 100%;
  height: 123px;
`;

const Caption = styled.div`
  padding-top: 50px;
  color: white;
  font-size: 13px;
  font-family: 'Montserrat', sans-serif;
  font-weight: 200;
  text-align: center;
  vertical-align: middle;
`;

class Modal extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      data: props.data,
      currentPhoto: 0,
    };
    this.closeModal = this.closeModal.bind(this);
    this.previousImage = this.previousImage.bind(this);
    this.nextImage = this.nextImage.bind(this);
  }

  closeModal(event) {
    this.props.closeModal && this.props.closeModal(event);
  }

  previousImage() {
    if (this.state.currentPhoto !== 0) {
      this.setState({ currentPhoto: this.state.currentPhoto - 1 });
    }
  }

  nextImage() {
    if (this.state.currentPhoto !== this.state.data.length - 1) {
      this.setState({ currentPhoto: this.state.currentPhoto + 1 });
    }
  }

  render() {
    if (!this.props.show) {
      return null;
    }
    const url = `${this.state.data[this.state.currentPhoto].url}`;
    const caption = `${this.state.data[this.state.currentPhoto].caption}`;
    return (
      <div>
        <Body>
          <Container>
            <TopBar>
              <Exit onClick={this.closeModal}>X</Exit>
            </TopBar>
            <Photo>
              <Left onClick={this.previousImage}>L</Left>
              <Image src={url} alt={caption} />
              <Right onClick={this.nextImage}>R</Right>
            </Photo>
            <BottomBar>
              <Caption>
                {`${this.state.currentPhoto + 1}/${this.state.data.length}:`}  {caption}
              </Caption>
            </BottomBar>
          </Container>
        </Body>
      </div>
    );
  }
}

export default Modal;
