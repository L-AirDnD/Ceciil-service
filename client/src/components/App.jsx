import React from 'react';
import styled from 'styled-components';
import Modal from './Modal.jsx';
import LargeImage from './MainLargeImage.jsx';
import SmallImages from './MainSmallImages.jsx';
import $ from 'jquery';
import { dummyData } from '../../../data';

const Parent = styled.div`
  display: flex;
  width: 100%;
  height: 430px;
`;

const Button = styled.button`
  position: absolute;
  top: 390px;
  right: 24px;
  padding: 6px 15px 6px 15px;
  color: #484848;
  border-radius: 4px;
  font-size: 14px;
  cursor: pointer;
`;

class Photos extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      data: dummyData[0],
      show: false,
    };
    this.showModal = this.showModal.bind(this);
  }

  componentDidMount() {
    $.ajax({
      url: '/api/photos/:offering',
      type: 'GET',
      success: (data) => {
        this.setState({ data: data });
      },
      error: () => {
        console.log('There was an error in your AJAX request...')
      }
    });
  }

  showModal() {
    this.setState({
      show: !this.state.show,
    });
  }

  render() {
    if (this.state.show) {
      return (
        <Modal data={this.state.data.photos} closeModal={this.showModal} show={this.state.show} />
      );
    }
    return (
      <div>
        <Parent>
          <LargeImage data={this.state.data.photos[0]} />
          <SmallImages data={this.state.data.photos.slice(1, 5)} />
          <Button onClick={this.showModal}>View Photos</Button>
        </Parent>
      </div>
    );
  }
}

export default Photos;
