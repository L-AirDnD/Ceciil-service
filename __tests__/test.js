import React from 'react';
import Enzyme, { mount, render, shallow} from 'enzyme';
import App from '../client/src/components/App.jsx';
import Modal from '../client/src/components/Modal.jsx';
import Adapter from 'enzyme-adapter-react-16';

Enzyme.configure({adapter: new Adapter});

describe('App', () => {
  test('should render', () => {
    const wrapper = shallow(<App />);
    expect(wrapper.exists()).toBe(true);
  });
  test('should not show modal upon rendering', () => {
    const wrapper = shallow(<App />);
    const showState = wrapper.state().show;
    expect(showState).toEqual(false);
  });
});

describe('Modal', () => {
  test('should render modal on button click', () => {
    const wrapper = mount(<App />);
    wrapper
      .find('button')
      .simulate('click');
    expect(wrapper).toMatchSnapshot();
    wrapper.unmount;
  });
  test('should render previous photo by clicking L', () => {
    const wrapper = shallow(<Modal />);
    const left = wrapper.find('Left');
    left
      .simulate('click')
    expect(left).to.have.length(1);
    expect(left).toMatchSnapshot();
  })
  test('should render next photo by clicking R', () => {
    const wrapper = shallow(<Modal />);
    const right = wrapper.find('Right');
    right
      .simulate('click')
    expect(right).to.have.length(1);
    expect(right).toMatchSnapshot();
  })
});
