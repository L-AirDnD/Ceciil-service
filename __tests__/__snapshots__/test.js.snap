// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal should render modal on button click 1`] = `ReactWrapper {}`;
